src/Dont/Exception/NonStringableObject.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dont\Exception;
+
+use LogicException;
+
+class NonStringableObject extends LogicException implements ExceptionInterface
+{
+    private const ERROR_TEMPLATE = <<<'ERROR'
+The given object %s#%s is not designed to be stringable.
+
+You tried to access a method called "__toString()".
+ERROR;
+
+    /**
+     * @param object $object
+     *
+     * @return NonStringableObject
+     *
+     * @throws TypeError
+     */
+    public static function fromAttemptedToString($object) : self
+    {
+        if (! is_object($object)) {
```

Comment:
How can this particular scenario be reached? Also, contradiction between docblock and this check

Reply:
tbh I copy pasted this from DontClone

Reply:
Removed https://github.com/Roave/Dont/pull/19/commits/cfb3dda63711a6a967b244f7766ae38b1b0adc14